src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -261,6 +284,45 @@ class RegularSync(
     }
   }
 
+  private def requestMissingNode(nodeId: ByteString): Unit = {
+    bestPeer match {
+      case Some(peer) =>
+        waitingForActor = Some(context.actorOf(
+          PeerRequestHandler.props[GetNodeData, NodeData](
+            peer, peerResponseTimeout, etcPeerManager, peerEventBus,
+            requestMsg = GetNodeData(List(nodeId)),
+            responseMsgCode = NodeData.code)))
+
+      case None =>
+        log.debug("Requesting missing state nodes: no peers to download from")
+        scheduleResume()
+    }
+  }
+
+  private def handleRedownloadedStateNodes(peer: Peer, nodes: Seq[ByteString]): Unit = {
+    if (nodes.isEmpty) {
+      log.debug(s"Did not receive missing state node from peer ($peer)")
+      resumeWithDifferentPeer(peer, "did not receive missing state node")
```

Comment:
if our best peer doesn't have a specific MPT node (that might be old) we will be blacklisting it even if best peer is selected by total difficulty (nothing related to state nodes). That being said, if you think that should be handled in another task or even dismissed as it doesn't present any problems or for this patch is good enough, dismiss this comment.

Reply:
It's not about about a peer not having this node, it simply about it refusing to give it to us. If the peer does not have the node we're most likely screwed anyway - we're too far behind, pruning has kicked in other nodes, and there's no way for us to recover.

Reply:
Peer doesn't have it or is refusing but it's giving us a lot of "good blocks". A good example is a good and responsive node that is running pruning (as the missing MPT node might be from any block within the history, right?). That being said, this is starting to become a philosophical discussion and this is meant to be a _patch_ so, lt's dismiss this comment.

Reply:
FWIW, blacklisting is only temporary, so by the time we receive the node and re-execute the blocks, the peer will most likely be un-blacklisted.